### PR TITLE
Scaffolding to start Wing IDE env (+ config_class.py, for refactoring config.py files)

### PR DESCRIPTION
**Pull request by HogJonny-AMZN**
Signed-off-by: HogJonny-AMZN <gallowj@amazon.com>

## What does this PR do?
Prototype Wing IDE integration for Technical Artists
- There is now a base class module for config.py files config_class.py,
- - this will streamline writing new config for DCC tools and IDEs (not all configs refactored in this PR)
- - upgrading pre-existing config.py modules is not in this scope of work (next iteration.)
- - only what is needed to get to wing in the most modern way is the critical path for this PR
- This class is used for the python script driven setup (wing/config.py)
- Wing support upgraded to Wing Pro 8, Wing 7x was still on py2.7, however I left the legacy .bat files in working order
- Some changes were made to the .bat env to accommodate the update
- It is verified that Wing Pro 8 will launch, start the interpreter, etc.

_Please add links to any issues, RFCs or other items that are relevant to this PR._
WIP work for GHI: https://github.com/o3de/o3de/issues/10526
WIP config base class, GHI: https://github.com/o3de/o3de/issues/10547

Notes:
- There was a Wing 8 bug where env slugs for python launch configurations were not resolving. I engaged their community via forums and requested a fix: https://ask.wingware.com/question/6588/in-launch-configurations/

## How was this PR tested?
- Wing 7x will launch with the legacy win .bat file env/launchers
- - i've managed to maintain 7x support for now, will deprecate in a future iteration: 
- -"DccScriptingInterface\Tools\Dev\Windows\IDE\Launch_WingPro_7x.bat"
- Added a wing 8 version there (will deprecate in the future):
- - "DccScriptingInterface\Tools\Dev\Windows\IDE\Launch_WingPro_8x.bat"

- Wing Pro 8 will launch and operate correct with the latest:
- "DccScriptingInterface\Tools\IDE\WingIDE\win_launch_wingide.bat"

- Wing Pro 8 will start from Windows CMD:

> cd o3de\Gems\AtomLyIntegration\TechnicalArt\DccScriptingInterface
> python.cmd Tools\IDE\Wing\start.py

### CODE REVIEW

**Review by shawstar — APPROVED** [review]

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/Dev/Windows/Env_DCC_Blender.bat [hunk 1]
```diff
@@ -34,7 +34,13 @@ IF "%DCCSI_BLENDER_VERSION%"=="" (set DCCSI_BLENDER_VERSION=3.1)
 :: Initialize env
 CALL %~dp0\Env_O3DE_Core.bat
 CALL %~dp0\Env_O3DE_Python.bat
-CALL %~dp0\Env_O3DE_Qt.bat
+
+:: This can now only be added late, in the launcher
+:: it conflicts with other Qt apps like Wing Pro 8+
+::CALL %~dp0\Env_O3DE_Qt.bat
+:: this could interfer with standalone python apps/tools/utils that use O3DE Qt
```
Comment (cgalvan) [inline Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/Dev/Windows/Env_DCC_Blender.bat:41]: Nitpick: `interfer` -> `interfere`

**Review by cgalvan — APPROVED** [review]

File: Registry/AssetProcessorPlatformConfig.setreg [hunk 2]
```diff
@@ -39,7 +39,7 @@
                     "StartupTimeoutSeconds" : 900
                 },
                 "Platform pc": {
-                    "tags": "tools,renderer,dx12,vulkan,null"
+                    "tags": "tools,renderer,dx12"
```
Comment (lumberyard-employee-dm) [inline Registry/AssetProcessorPlatformConfig.setreg:42]: Is this change intended?
Doesn't this cause processing assets targeting vulkan and the null renderer from the AP to fail?

Comment (HogJonny-AMZN) [reply Registry/AssetProcessorPlatformConfig.setreg:42]: yes, this was mistakenly added

**Review by shawstar — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]
